zapcore/write_syncer_test.go

```diff
@@ -65,6 +71,80 @@ func TestAddSyncWriter(t *testing.T) {
 	assert.NoError(t, ws.Sync(), "Unexpected error calling a no-op Sync method.")
 }
 
+func TestBufferWriter(t *testing.T) {
+	defer goleak.VerifyNone(t)
+
+	// If we pass a plain io.Writer, make sure that we still get a WriteSyncer
+	// with a no-op Sync.
```

Comment:
The tests don't validate that the buffer writer doesn't acutally write to the underlying buffer straight away. We should ideally have a test that writes a log that should be significantly smaller than the buffer, and it should not be available in the underlying writer till after an explicit Sync or the background ticker flushes